cmd/livepeer_cli/wizard_ticketbroker.go

```diff
@@ -58,8 +58,8 @@ func (w *wizard) deposit() {
 	reserveAmount := w.readPositiveFloat()
 
 	form := url.Values{
-		"depositAmount": {eth.ToBaseUnit(big.NewFloat(depositAmount)).String()},
-		"reserveAmount": {eth.ToBaseUnit(big.NewFloat(reserveAmount)).String()},
+		"depositAmount": {eth.EthToWei(depositAmount).String()},
```

Comment:
Instead of converting the user's string input into a float can we skip the float conversion and just use the string's integer and decimal components to convert into a `big.Int` which validity checks can be applied against?

Something like:

```
// in -> user string input
parts := strings.Split(in, ".")
// parts[0] -> integer component
// parts[1] -> decimal component

if len(parts[1]) > 18 {
    // Cannot have more than 18 decimal places
}

val, ok := new(big.Int).SetString(parts[0] + parts[1])
// Check that conversion worked

// val -> string input represented in base units

// val.String() can be used to return a string when creating a request
```

Here is an [example](https://github.com/filecoin-project/go-filecoin/blob/b926dad7ba3d5a6342d73008d465c2a56a37d04e/internal/pkg/types/atto_fil.go#L87) from another codebase. 

We could update the `readPositiveFloat()` helper to return a `big.Int` using the above approach. Might want to update the name of the relevant methods to something more accurate like `readTokenUnitString` and readPositiveTokenUnitString()`.

Reply:
I would prefer to not introduce a third party library such as as is, so therefore I like an alternative that just uses the standard library. Just wondering if depending on a string.Split() here is like a robust enough solution, seems so simplistic (not saying that it's not good).

Reply:
There is one thing I don't like though but I don't think it's that important _for now_ as the only usage is comparing the returned value to 0.01 which will always be greater than the point where we start rounding. 

When converting a wei string to eth we will eventually end up converting to a `*big.Float` most likely, whether we return the value as a string or let the the caller handle conversion. This will always end up resulting in rounding errors as this failing test shows.

```
        	Error Trace:	helpers_test.go:78
        	Error:      	Not equal: 
        	            	expected: "0.003333333333"
        	            	actual  : "0.003333333333333333"
```

Reply:
Not sure I understand. I'm suggesting that we convert all user string input directly into `big.Int` - we would never use floats at all. If the user input is in ETH, then the string input would be parsed and represented in wei using a `big.Int`.

Reply:
Right but in `FormatUnits` we convert a wei string to an eth value as `big.Float` (which is the other way around of what's happening with the CLI input for deposits and reserves). 

Converting to a big float there is okay for now because we don't need more precision than 2 decimals, as this is the only place where `FromBaseUnit` or in this PR `WeiToEth` is used.

```
func FormatUnits(baseAmount *big.Int, name string) string {
	if baseAmount == nil {
		return "0 " + name
	}
	
	amount := FromBaseUnit(baseAmount)

	if amount.Cmp(big.NewFloat(0.01)) == -1 {
		switch name {
		case "ETH":
			return fmt.Sprintf("%v WEI", baseAmount)
		default:
			return fmt.Sprintf("%v LPTU", baseAmount)
		}
	} else {
		switch name {
		case "ETH":
			return fmt.Sprintf("%v ETH", amount)
		default:
			return fmt.Sprintf("%v LPT", amount)
		}
	}
}
```

Reply:
Oh I see what you're saying now.

We could do the following:

- Update `ToBaseUnit(*big.Float) *big.Int)` to `ToBaseUnit(string) (*big.Int, bool)`
- Update `FromBaseUnit(*big.Int) *big.Float)` to `FromBaseUnit(*big.Int) string)`
- Update `FormatUnits()` to compare `baseAmount` against a `big.Int` instead of converting `baseAmount` to `big.Float` and comparing it against a `big.Float`

`ToBaseUnit()` could be used by the CLI to handle user inputs denominated in ETH.

Reply:
submitted several fixup commits